src/state.py

```diff
@@ -11,6 +12,69 @@ def __init__(self):
     def add_student(self, student):
         self.students[student.student_id] = student
 
+    def get_score(self): 
+        score = 0
+        #if traded the subject as expected get score
+        for i in self.students.values(): 
+
+            #FLAGS 
+            alone = True        #the person is not with a friend
+            no_givin = True     #the person didn't givin any class 
+            no_target = True    #the person didn't get the target 
+
+            #checking buddies 
+            for numbers in set(list(i.buddies.values())):
+                    for classes in set(list(i.buddies.keys())): 
+                        if i.subjects_and_classes[classes] == self.students[numbers].subjects_and_classes[classes]: 
+                            score += 30     #the person is at the same class of his friend
+                            alone = False   #Flag-- 
```

Comment:
In each loop of checking, flags are updated, in order to avoid creating other loops.